external-crates/move/crates/move-package-alt/src/package/package_impl.rs

```diff
@@ -230,36 +230,53 @@ impl<F: MoveFlavor> Package<F> {
-}
 
-/// Return the implicit deps depending on the implicit dep mode. Note that if `implicit_dep_mode`
-/// is ImplicitDepMode::Legacy, a `legacy_manifest` is required otherwise it will panic.
-// TODO this needs to be moved into a ImplicitDeps trait
-fn implicit_deps<F: MoveFlavor>(
-    env: &Environment,
-    implicit_dep_mode: ImplicitDepMode,
-    legacy_manifest: Option<&ParsedLegacyPackage>,
-) -> BTreeMap<PackageName, ReplacementDependency> {
-    // TODO: is this correct? Do we need to include the git repo? What if somebody names it
-    // differently and specifies a different git repository (e.g., a fork) - is that possible?
-    // e.g., SuiFork = { git = "https://github.com/suifork/sui.git", branch = "main" }
-
-    match implicit_dep_mode {
-        ImplicitDepMode::Enabled => F::implicit_deps(env.id().to_string()),
-        ImplicitDepMode::Disabled => BTreeMap::new(),
-        ImplicitDepMode::Legacy => {
-            let system_deps_in_legacy_manifest = legacy_manifest
-                .expect("Legacy manifest should be present")
-                .deps
-                .iter()
-                .any(|(name, _)| SYSTEM_DEPS_NAMES.contains(&name.as_str()));
-            // if the legacy manifest has system dependencies, we return an empty map
-            if system_deps_in_legacy_manifest {
-                BTreeMap::new()
-            } else {
-                F::implicit_deps(env.id().to_string())
+    /// Return the implicit deps depending on the implicit dep mode. Note that if `implicit_dep_mode`
+    /// is ImplicitDepMode::Legacy, a `legacy_manifest` is required otherwise it will panic.
+    // TODO this needs to be moved into a ImplicitDeps trait
+    fn implicit_deps(
+        env: &Environment,
+        implicit_dep_mode: ImplicitDepMode,
+        legacy_manifest: Option<&ParsedLegacyPackage>,
+    ) -> BTreeMap<PackageName, ReplacementDependency> {
+        // TODO - rethink how this implict dep mode works
+        // let system_pacakages = F::implicit_deps;
+        // if let Some(legacy) = package.legacy_deta {
+        //   check if package is a system package (i.e. if package.name() is in system_packages)
+        //   check if package has any deps with same names as system packages
+        //   if either is true: set implicit_dep_mode to Disabled
+        // }
+
+        match implicit_dep_mode {
+            // for modern packages, the manifest should have the implicit-deps field set to true
+            // (by default), or to false.
+            ImplicitDepMode::Enabled => F::implicit_deps(env.id().to_string()),
+            ImplicitDepMode::Disabled => BTreeMap::new(),
+            ImplicitDepMode::Legacy => {
+                let legacy_manifest = legacy_manifest.expect("Legacy manifest should be present");
+                let system_deps_in_legacy_manifest = legacy_manifest
+                    .deps
+                    .iter()
+                    .any(|(name, _)| SYSTEM_DEPS_NAMES.contains(&name.as_str()));
```

Comment:
this will need reworking, but seems ok for now. I assume `SYSTEM_DEPS_NAMES` contains the upper-cased names?